### PR TITLE
Don't crash when receiving a bad uri via send tab

### PR DESCRIPTION
**Pull request by skhamis**
fixes https://github.com/mozilla-mobile/fenix/issues/20414 

Ideally this shouldn't have been possible since clients should be not allowing certain URI schemes from being send-capable. With synced tabs we follow this pref: https://searchfox.org/mozilla-central/source/modules/libpref/init/all.js#4372 which I carried over and inputted for android. 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/NotificationManager.kt
Comment (mcarare) [inline app/src/main/java/org/mozilla/fenix/components/NotificationManager.kt:137]: Are we planning to use this just as a workaround until the main cause is fixed? (i.e. not syncing these schemas?)

Comment (skhamis) [reply app/src/main/java/org/mozilla/fenix/components/NotificationManager.kt:137]: I think this would be a semi-permanent solution as it could be a very long-tail to ensure no version at some point could send a file:// across all browser platforms. Though I do think it's a relatively straightforward solution to just copy what desktop is doing for now but happy to brainstorm a different approach if we feel strongly against this.

**Comment by skhamis** [comment]
Hi @mcarare, how are we feeling about this PR? Anything I need to do to continue moving this one along?

**Comment by mcarare** [comment]
@skhamis The solution presented here is ok. I just feel the fix should not be needed at all if the problem is fixed at its origin.

**Comment by skhamis** [comment]
> @skhamis The solution presented here is ok. I just feel the fix should not be needed at all if the problem is fixed at its origin.

Totally understand the concern @mcarare! While the team will indeed fix underlying issue of having `file://` be an option for send push on Desktop, iOS and Android. This solution should [prevent current users from crashing](https://github.com/mozilla-mobile/fenix/issues/20414) since it'll take quite awhile (most likely months) before those fixes go out to users and we reach a point where users can't send invalid schemes. 

Alternatively, we can indeed just sit/close this PR for now and wait until the send-tab fixes land on the respective repos -- unfortunately, users will still be crashing if they send invalid schemas until then but depending on the need/want to have that I'll leave to the Fenix team!

**Comment by mhammond** [comment]
> @skhamis The solution presented here is ok. I just feel the fix should not be needed at all if the problem is fixed at its origin.

@mcarare further to what @skhamis said, the problem isn't just that the fix is only on Nightly (and thus will not be on release for a couple of months), there's a very long tail of desktop versions - not only is there ESR users, it's not that uncommon to find desktop users 10 versions or so behind. So while this is your call, you should expect that it will be *at least* 6 months before you can be confident that remote Firefoxes crashing Fenix is rare.

**Review by mcarare — APPROVED** [review]

**Pull request merged**
